### PR TITLE
atomic: Add support for gcc __sync intrinsics

### PR DESCRIPTION
**Pull request by Lastique**
This adds support for older gcc versions that didn't support __atomic intrinsics.

Related to https://github.com/baresip/re/pull/446#issuecomment-1204383834

CC @cspiel1

### CODE REVIEW

**Comment by cspiel1** [comment]
I will try this with our blackfin build system.

**Comment by cspiel1** [comment]
```
src/main/main.c: In function 're_main':
src/main/main.c:1070: error: expected ')' before ';' token
src/main/main.c:1094: error: expected ')' before ';' token
```
Something wrong here with this line:
```
	if (re_atomic_rlx(&re->polling)) {
```

File: include/re_atomic.h [hunk 1]
```diff
@@ -127,6 +127,323 @@
...
+	})
+
+#else
+
+#define re_atomic_store(_a, _v, _mo) \
+	(void)re_atomic_exchange(_a, _v, _mo)
+
+#define re_atomic_load(_a, _mo) \
+	__sync_val_compare_and_swap(\
+		_a, (__typeof__(*(_a)))0, (__typeof__(*(_a)))0);
```
Comment (cspiel1) [inline include/re_atomic.h:395]: The ':' at the end seems to be wrong

Comment (Lastique) [reply include/re_atomic.h]: Fixed, thanks.

**Comment by cspiel1** [comment]
Looks good now. It builds. Thanks!

**Pull request merged**